apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/preadj/R4ClaimResourceProviderIT.java

```diff
@@ -0,0 +1,142 @@
+package gov.cms.bfd.server.war.r4.providers.preadj;
+
+import static org.junit.Assert.assertEquals;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.param.DateParam;
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import com.google.common.collect.ImmutableMap;
+import gov.cms.bfd.model.rda.PreAdjFissClaim;
+import gov.cms.bfd.model.rda.PreAdjFissProcCode;
+import gov.cms.bfd.model.rda.PreAdjMcsClaim;
+import gov.cms.bfd.model.rda.PreAdjMcsDetail;
+import gov.cms.bfd.model.rda.PreAdjMcsDiagnosisCode;
+import gov.cms.bfd.server.war.ServerTestUtils;
+import gov.cms.bfd.server.war.utils.RDATestUtils;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Claim;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.Resource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class R4ClaimResourceProviderIT {
+
+  private static final RDATestUtils testUtils = new RDATestUtils();
+
+  @BeforeClass
+  public static void init() {
+    testUtils.init();
+
+    testUtils.seedData(testUtils.fissTestData());
+    testUtils.seedData(testUtils.mcsTestData());
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    testUtils.truncate(PreAdjFissProcCode.class);
+    testUtils.truncate(PreAdjFissClaim.class);
+    testUtils.truncate(PreAdjMcsDetail.class);
+    testUtils.truncate(PreAdjMcsDiagnosisCode.class);
+    testUtils.truncate(PreAdjMcsClaim.class);
```

Comment:
How about putting all of this cleanup into a single method in RDATestUtils since we will be adding additional tables as RDA adds more fields that require detail tables.  Or maybe we should add an ImmutableList of entity classes in a static final somewhere in bfd-model-rda to use?  Otherwise this sort of boilerplate could bite us later.